ASM/c/dungeon_info.c

```diff
@@ -293,12 +295,36 @@ void draw_dungeon_info(z64_disp_buf_t* db) {
 
         left += icon_size + padding;
 
-        // Draw dungeon names
-
+        // First draw precompleted dungeons, greyed out with a small rectangle to cross them.
+        gDPSetPrimColor(db->p++, 0, 0, 0xFF, 0xFF, 0xFF, 0xBF);
```

Comment:
So, why not here split this out into two functions, one that draws the dungeon name white and one the draws it grey with the white cross-out rectangle over it. Then just iterate over the dungeons once and call those functions depending on if it is a pre-completed dungeon or not.

Reply:
So there is actually a reason i wrote it like that because i agree it is very much not elegant.
Calling too many text_flush in a row make the game crash, i guess because the main DL buffer is overflowing.
It probably loads too many sprites at once or something.
So the idea was to minimize the text_flush calls, and so i had to do two loops.
But yeah maybe we should fix instead this text_flush function..

Reply:
I see, it loads a bunch of new textures every time it is called. It should probably load a font texture once in `text_init` and then use `gSPTextureRectangle` to use parts of the texture for each character like GfxPrint does.